### ISSUE TITLE
assure that repeated errors have unique wording

### ISSUE THREAD

**Issue by praiskup** [PyCQA/pylint#3508]
Short story, would it be possible to describe the problem more concretely so
different instances of the same report don't have the same wording?  E.g.

```diff
- Too few public methods (0/2)
+ Too few public methods in ClassName (0/2)
```

Long story, ... before merging any PR, we want to to generate two sets of pylint
issues (before and after applying the patches from PR) and then diff them.  Only
the errors which are new should be fixed before we merge the code.

For generic "static analysis report" diff'ing, there's [csdiff](https://github.com/kdudka/csdiff)
project.  It runs as `csdiff old-report.log new-report.log` and it only
keeps the added (new) issues on it's stdout.

Csdiff ignores line numbers in reports, that's because the code moves all
the time within the same file.

The thing is that in [some situations](https://github.com/kdudka/csdiff/issues/6),
multiple occurrences of the similar issues exist in the same file and
csdiff isn't able to distinguish between them.

**Comment by praiskup** [comment]
Wrong, that's what `{obj}` is for in `--msg-template`.  Sorry for noise.
